spacy/training/loggers.py

```diff
@@ -23,13 +26,35 @@ def setup_table(
     return final_cols, final_widths, ["r" for _ in final_widths]
 
 
-@registry.loggers("spacy.ConsoleLogger.v1")
-def console_logger(progress_bar: bool = False):
+@registry.loggers("spacy.ConsoleLogger.v2")
+def console_logger(
+    progress_bar: bool = False,
+    print_progress: bool = True,
+    output_file: Union[str, Path] = None,
+):
+    """The ConsoleLogger.v2 prints out training logs in the console and saves them into a jsonl file, with both options being optional.
+    print_progress (bool): Whether the logger should print the logs.
+    progress_bar (bool): Whether the logger should print the progress bar.
+    output_file (Union[str, Path]): Where the logger should save the training logs to.
+    """
+    if output_file:
+        output_file = util.ensure_path(output_file)
+        if output_file.is_file():
+            output_file = util.ensure_path(output_file.stem + ".jsonl")
+            if output_file.exists():
+                output_file.unlink()
+        if not output_file.parents[0].exists():
+            output_file.parents[0].mkdir(parents=True)
+
     def setup_printer(
         nlp: "Language", stdout: IO = sys.stdout, stderr: IO = sys.stderr
     ) -> Tuple[Callable[[Optional[Dict[str, Any]]], None], Callable[[], None]]:
         write = lambda text: print(text, file=stdout, flush=True)
         msg = Printer(no_print=True)
+        if not print_progress:
+            write(msg.info(f"Printing to console is disabled"))
```

Comment:
I don't think we should write anything here, if users turn console output off then we shouldn't print further warnings on the console IMO